src/Tokenizer/Token.php

```diff
@@ -156,34 +156,44 @@ public function clearChanged()
      */
     public function equals($other, $caseSensitive = true)
     {
-        $otherPrototype = $other instanceof self ? $other->getPrototype() : $other;
+        if ($other instanceof self) {
+            if ($this->isArray !== $other->isArray) {
+                return false;
+            }
+
+            if ($this->id !== $other->id || $this->content !== $other->content) {
+                return false;
+            }
+
+            return true;
+        }
 
-        if ($this->isArray !== \is_array($otherPrototype)) {
+        if ($this->isArray !== \is_array($other)) {
             return false;
         }
 
         if (!$this->isArray) {
-            return $this->content === $otherPrototype;
+            return $this->content === $other;
         }
 
-        if (array_key_exists(0, $otherPrototype) && $this->id !== $otherPrototype[0]) {
+        if (\array_key_exists(0, $other) && $this->id !== $other[0]) {
```

Comment:
please revert backslashing.
we rely on `native_function_invocation` rule, that's it

Reply:
this path is very very hot, and paying this additional namespace lookup here makes a difference.

Reply:
How much difference?

Reply:
after https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4041 its no longer measurable.